packages/0xcert-ethereum-generic-provider/src/core/provider.ts

```diff
@@ -199,17 +199,25 @@ export class GenericProvider extends EventEmitter implements ProviderBase {
   }
 
   /**
-   * Returns order gateway ID (address).
+   * Returns gateway config.
    */
-  public get orderGatewayId(): string {
-    return this._orderGatewayId || null;
+  public get gatewayConfig(): GatewayConfig {
+    return this._gatewayConfig || null;
   }
 
   /**
-   * Sets and normalizes account ID.
+   * Sets and normalizes gateway config.
    */
-  public set orderGatewayId(id: string) {
-    this._orderGatewayId = this.encoder.normalizeAddress(id);
+  public set gatewayConfig(config: GatewayConfig) {
+    if (typeof config !== 'undefined') {
+      this._gatewayConfig = {
+        multiOrderId: this.encoder.normalizeAddress(config.multiOrderId),
+        assetLedgerDeployOrderId: this.encoder.normalizeAddress(config.assetLedgerDeployOrderId),
+        valueLedgerDeployOrderId: this.encoder.normalizeAddress(config.valueLedgerDeployOrderId),
+      };
+    } else {
+      this._gatewayConfig = null;
```

Comment:
We are not throwing an error in any other case and I cannot set to `{}` since in that case it demands for object properties to be set. We also return null now for the case of `accountId`.

Reply:
Let's think about how we can do this part easier for a user. We can maybe add a methods like `Provider.getRopstenGatewayConfig()` and `Provider.getLiveGatewayConfig()` which would allow you to do this
```ts
new Provider({
  gatewayConfig: Provider.getRopstenGatewayConfig();
  ...
});
```